### PR TITLE
Consistent versioning

### PR DESCRIPTION
**Pull request by samizzy**
Change Avro versioning to semantic versioning so that the version numbers are consistent across different schema types.

### CODE REVIEW

File: api-metastore/src/main/java/org/zalando/nakadi/service/SchemaEvolutionService.java [hunk 1]
```diff
@@ -109,18 +107,25 @@ private void collectErrorMessages(final ValidationException e,
...
         checkEvolutionIncompatibilities(original, eventType);
         final var version = original.getSchema().getVersion();
+        final var typeChanged = !original.getSchema().getType().
+                equals(eventType.getSchema().getType());
+
+        if(typeChanged) {
+            return bumpVersion(original, eventType, MAJOR, new Version(version));
+        }
+
         switch (eventType.getSchema().getType()) {
```
Comment (adyach) [inline api-metastore/src/main/java/org/zalando/nakadi/service/SchemaEvolutionService.java:117]: why do we need `switch` if it looks the same for any `type` ?

Comment (samizzy) [reply api-metastore/src/main/java/org/zalando/nakadi/service/SchemaEvolutionService.java:117]: The parameters are same but the methods are different. `evolveJsonSchema()` vs `evolveAvroSchema()`

Comment (adyach) [reply api-metastore/src/main/java/org/zalando/nakadi/service/SchemaEvolutionService.java:117]: ups, missed that

**Comment by adyach** [comment]
👍

**Comment by samizzy** [comment]
:+1:

**Pull request merged**